src/c_api/c_api.cc

```diff
@@ -497,12 +499,139 @@ XGB_DLL int XGDMatrixCreateFromMat_omp(const bst_float* data,  // NOLINT
       }
     }
   }
+  // restore omp state
+  omp_set_num_threads(nthread_orig);
 
   mat.info.num_nonzero_ = mat.page_.data.size();
   *out  = new std::shared_ptr<DMatrix>(DMatrix::Create(std::move(source)));
   API_END();
 }
 
+int DTGetType(const wchar_t* type_string) {
+  if (wcscmp(type_string, L"float32") == 0) {
+    return 0;
```

Comment:
I'm nervous about returning magic numbers here, as the return value of `DTGetType()` will be directly used by `DTGetValue()`. Can we create an enum class to create a group of integer constants? Something like
```cpp
enum class DTType : uint8_t {
  kFloat32 = 0,
  kFloat64 = 1,
  kBool8 = 2,
  kInt32 = 3,
  kInt8 = 4,
  kInt16 = 5,
  kInt64 = 6
};
```
Then the `DTGetType()` function will be like
```cpp
DTType DTGetType(const wchar_t* type_string) {
  if (wcscmp(type_string, L"float32") == 0) {
    return DTType::kFloat32;
  }  else if (wcscmp(type_string, L"float64") == 0) {
    return DTType::kFloat64;
  } /* ... other types ... */
}
```
and the `DTGetValue()` will be like
```cpp
float DTGetValue(void* column, DTType dt_type, size_t ridx) {
  float missing = std::numeric_limits<float>::quiet_NaN();
  switch (dt_type) {
     case DTType::kFloat32: { /* ... */ }
     case DTType::kFloat64: { /* ... */ }
     /* ... other types ... */
  }
}
```